pkg/client/parse.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/guidewire-oss/fern-junit-client/pkg/models/fern"
+	"github.com/guidewire-oss/fern-junit-client/pkg/models/junit"
+)
+
+func parseReports(testRun *fern.TestRun, filePattern string, verbose bool) error {
+	files, err := filepath.Glob(filePattern)
+	if err != nil {
+		return fmt.Errorf("failed to parse file pattern %s: %w", filePattern, err)
+	}
+	if len(files) == 0 {
+		return fmt.Errorf("no files found for pattern %s", filePattern)
+	}
+	for _, file := range files {
+		suiteRun, err := parseReport(file, verbose)
+		if err != nil {
+			return fmt.Errorf("failed to parse report %s: %w", file, err)
+		}
+		testRun.SuiteRuns = append(testRun.SuiteRuns, suiteRun...)
+	}
+	// TODO: set testRun.StartTime to the earliest suite start time
+	// TODO: set testRun.EndTime to the latest suite end time
```

Comment:
I created an issue for this: https://github.com/guidewire-oss/fern-junit-client/issues/4